File: packages/types/src/issues/worklog.ts
/**
 * Copyright (c) 2023-present Plane Software, Inc. and contributors
 * SPDX-License-Identifier: AGPL-3.0-only
 * See the LICENSE file for details.
 */

// [FA-CUSTOM] Types for time tracking / worklogs

export type TIssueWorklog = {
  id: string;
  issue: string;
  project: string;
  workspace: string;
  actor: string;
  actor_detail: {
    id: string;
    first_name: string;
    last_name: string;
    display_name: string;
    avatar_url: string;
  };
  duration_minutes: number;
  date_worked: string; // "YYYY-MM-DD"
  description: string;
  created_at: string;
  updated_at: string;
  created_by: string;
};

export type TIssueWorklogMap = Record<string, TIssueWorklog>;

export type TIssueWorklogIdMap = Record<string, string[]>;

File: apps/api/plane/app/serializers/worklog.py
# Copyright (c) 2023-present Plane Software, Inc. and contributors
# SPDX-License-Identifier: AGPL-3.0-only
# See the LICENSE file for details.

# [FA-CUSTOM] Serializer for IssueWorklog (time tracking)

from .base import BaseSerializer
from .user import UserLiteSerializer
from plane.db.models import IssueWorklog


class IssueWorklogSerializer(BaseSerializer):
    actor_detail = UserLiteSerializer(read_only=True, source="actor")

    class Meta:
        model = IssueWorklog
        fields = "__all__"
        read_only_fields = [
            "workspace",
            "project",
            "issue",
            "created_by",
            "updated_by",
            "created_at",
            "updated_at",
        ]
